client/jetpack-cloud/sections/agency-signup/agency-signup-form/index.tsx

```diff
@@ -75,9 +75,12 @@ export default function AgencySignupForm() {
 		[ notificationId, partner?.id, createPartner.mutate, dispatch ]
 	);
 
-	// Redirect the user if they are already a partner or the form was submitted successfully.
+	// Redirect the user to the dashboard if they are already a partner,
+	// or the overview page if the form was submitted successfully.
 	useEffect( () => {
-		if ( partner ) {
+		if ( createPartner.isSuccess ) {
+			page.redirect( overviewPath() );
+		} else if ( partner ) {
```

Comment:
This would mean we start redirecting agencies to the new (empty) overview page as soon as we merge this PR.

We should separate this into a future diff when we're ready for this specific step.

Reply:
Good point...

Reply:
d9c7c89